### PR TITLE
Enable GPU CI on GitHub Actions self-hosted runner

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

- This PR enables GPU CI using legacy CUDA on deterministic tests running on sulfur as a self-hosted runner on GitHub Actions.
- Set new workflow configuration file to trigger with current on-demand workflow: "Test this please" as PR comment by authorized user. Currently the new status will only show up if triggered.
- Set LD_LIBRARY_PATH to point at CUDA runtime libraries

To verify/understand:
- [ ] YAML GitHub Actions structure and functionality
- [ ] Make sure PR status is reported as a check when triggered 
- [ ] Directory structure for jobs on sulfur as a self-hosted runner (currently enabled)
- [ ] Status state is correctly reported through `${{job.status}}`
- [ ] Only authorized users can trigger the workflow

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature (GPU CI test)
- Testing changes (new CI workflow to test with `-DQMC_CUDA=1` on self-hosted runner)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
sulfur locally, entire workflow must run on GitHub Actions

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
I don't like exposing ci-github-actions-self-hosted.yaml in this public repo. Can we move it somehow or put the more delicate parts in a private repo different from this one?

**Review by prckent — CHANGES_REQUESTED** [review]
Comments left above. Appropriately this is [WIP].

**Comment by williamfgc** [comment]
Hi @prckent thanks for the feedback. As we discussed offline, we're using GitHub secrets to address your valid concern. Now we have a chicken-and-egg problem since this needs to be merged in order to be fully tested on CI with a "dummy" PR. Let me know if you other points to address.

**Review by prckent — DISMISSED** [review]

**Comment by prckent** [comment]
I added a mention of the precision in the legacy CUDA configure line as a reminder for future readers that the default precision is mixed and different from the CPU default.

Remove the [WIP]?

**Comment by williamfgc** [comment]
@prckent please don't merge yet, I'm testing the conditional in another repo. This won't work as-is (among other things we'll find after merging).

**Comment by williamfgc** [comment]
@prckent OK, removed the [WIP]. The conditional is enabled through GitHub Actions `env` variables, secrets are hidden from logs and those `env` variables are not used anywhere in the workflows. After merge more testing is needed.

**Comment by williamfgc** [comment]
@ye-luo let me know if you have any feedback before merging and attempting to test the actual workflow. Thanks!

**Comment by ye-luo** [comment]
Test this please

**Comment by prckent** [comment]
The changes here are easily reversible if needed. @ye-luo OK to merge?

**Comment by ye-luo** [comment]
I don't see a new entry after saying the magic word. @williamfgc is this expected to work?

**Comment by williamfgc** [comment]
@ye-luo it should only be triggered if it's part of the `develop` branch, so a new PR won't change the expected behavior, from [docs](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issue_comment). Once merged we can do a dummy PR to test.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**
